sphinx/util/nodes.py

```diff
@@ -43,24 +47,22 @@ class NodeMatcher:
     and ``reftype`` attributes::
 
         matcher = NodeMatcher(nodes.reference, refdomain='std', reftype='citation')
-        doctree.findall(matcher)
+        matcher.findall_in(doctree)
         # => [<reference ...>, <reference ...>, ...]
 
     A special value ``typing.Any`` matches any kind of node-attributes.  For example,
     following example searches ``reference`` node having ``refdomain`` attributes::
 
-        from __future__ import annotations
-    from typing import TYPE_CHECKING, Any
         matcher = NodeMatcher(nodes.reference, refdomain=Any)
-        doctree.findall(matcher)
+        matcher.findall_in(doctree)
         # => [<reference ...>, <reference ...>, ...]
     """
 
-    def __init__(self, *node_classes: type[Node], **attrs: Any) -> None:
+    def __init__(self, *node_classes: type[T], **attrs: Any) -> None:
         self.classes = node_classes
         self.attrs = attrs
 
-    def match(self, node: Node) -> bool:
+    def _match(self, node: Node) -> bool:
```

Comment:
I think we should keep it public since people might rely on this class. The class is not really private IMO and extensions could potentially use it.

Reply:
done